lib/Onyx.d.ts

```diff
@@ -0,0 +1,278 @@
+import {Component} from 'react';
+import {PartialDeep} from 'type-fest';
+import * as Logger from './Logger';
+import {DeepRecord, Key, CollectionKey, Selector, Value} from './types';
+
+type KeyValueMap = {
+    [TKey in Key | CollectionKey]: Value[TKey];
+};
+
+type BaseConnectOptions<TKey extends Key | CollectionKey> = {
+    statePropertyName?: string;
+    withOnyxInstance?: Component;
+    initWithStoredValues?: boolean;
+    selector?: Selector<TKey, Value[TKey] | null>;
+};
+
+type ConnectOptions<TKey extends Key | CollectionKey> = BaseConnectOptions<TKey> &
+    (
+        | {
+              key: TKey extends CollectionKey ? TKey : never;
+              callback?: (value: Record<string, Value[TKey]> | null, key?: TKey) => void;
+              waitForCollectionCallback: true;
+          }
+        | {
+              key: TKey;
+              callback?: (value: Value[TKey] | null, key?: TKey) => void;
+              waitForCollectionCallback?: false;
+          }
+    );
+
+type MergeCollection<TKey extends Key, TMap, TValue> = {
+    [MapK in keyof TMap]: MapK extends `${TKey}${string}`
+        ? MapK extends `${TKey}`
+            ? never // forbids empty id
+            : TValue
+        : never;
+};
+
+type UpdateOperation<TKey extends Key | CollectionKey> =
+    | {
+          onyxMethod: 'set';
+          key: TKey;
+          value: Value[TKey];
```

Comment:
can this be possibly `null`?

Reply:
@hayata-suenaga Looking at the implementation it seems possible, will test and let you know.

Reply:
We definitely have updates / operations with each method, where the value is `null`. It's very common. `Value[TKey]` is `unknown` by default so I think this checks out.

Maybe I'm not understanding the question though.

Reply:
> We definitely have updates / operations with each method, where the value is null. It's very common.
@neil-marcellini thank you for the info 🙇 

> `Value[TKey]` is `unknown` by default 
`Value[TKey]` is actually the type you define for an onyx key. `TKey` is one of the onyx key and `Value[TKey]` is the value type for that key

Reply:
@hayata-suenaga @neil-marcellini Yeas we can pass `null`, will change it!